src/Shared/ParameterBindingMethodCache.cs

```diff
@@ -168,19 +174,17 @@ public bool HasTryParseMethod(ParameterInfo parameter)
             {
                 var hasParameterInfo = true;
                 // There should only be one BindAsync method with these parameters since C# does not allow overloading on return type.
-                var methodInfo = GetStaticMethodFromHierarchy(nonNullableParameterType, "BindAsync", new[] { typeof(HttpContext), typeof(ParameterInfo) });
-                if (methodInfo is null)
+                var methodInfo = GetStaticMethodFromHierarchy(nonNullableParameterType, "BindAsync", new[] { typeof(HttpContext), typeof(ParameterInfo) }, ValidateReturnType);
+                if (methodInfo is null || !ValidateReturnType(methodInfo))
```

Comment:
So if there is no `BindAsync` method or it doesn't produce the a valid return type then we fallback to `BindAsync(HttpContext)` without a parameter info. What is this functionally doing?

Reply:
> or it doesn't produce the a valid return type

I thought I removed that `||` statement, it's now passed directly to `GetStaticMethodFromHierarchy` so isn't needed anymore.

> What is this functionally doing?

Falling back from `BindAsync(HttpContext, ParameterInfo)` to `BindAsync(HttpContext)`.